### ISSUE TITLE
Bug: MultiWorld.can_beat_game with no arguments initially checks if the wrong state is beatable

### ISSUE THREAD

**Issue by Mysteryem** [ArchipelagoMW/Archipelago#3742]
### What happened?

It seemed odd that `MultiWorld.can_beat_game()` with no arguments would initially check if `self.state` is beatable, but then create an empty state, `state = CollectionState(self)`, to sweep spheres from to determine if the game is beatable, when these two states could be different:
https://github.com/ArchipelagoMW/Archipelago/blob/575c338aa3c895aa4d10824be5380b4e094b55e1/BaseClasses.py#L540C1-L547C42

This appears to be a bug introduced way back in https://github.com/ArchipelagoMW/Archipelago/commit/27b6dd8bd76125b5fe9c9b8cb3e1fc3be3af68bd

Judging by the original code in that commit, I suspect that
```py
        if starting_state:
            if self.has_beaten_game(starting_state):
                return True
            state = starting_state.copy()
        else:
            if self.has_beaten_game(self.state):
                return True
            state = CollectionState(self)
```
should have been:
```py
        if starting_state:
            if self.has_beaten_game(starting_state):
                return True
            state = starting_state.copy()
        else:
            state = CollectionState(self)
            if self.has_beaten_game(state):
                return True
            
```

It seems that everywhere in Archipelago's codebase that calls `MultiWorld.can_beat_game()` with no arguments or with `starting_state=None` has a `multiworld.state` (`self.state` in the above) that only contains precollected items, so `CollectionState(multiworld)` happens to result in an equivalent state, but this should not be relied on to always be the case.

### What were the expected results?

The state the sphere sweep starts from should be always be the state that is initially checked for being beatable, or a copy of it.

### Software

Local generation

**Comment by Exempt-Medic** [comment]
Is there a reason this isn't just a PR?

**Comment by Mysteryem** [comment]
Under the assumption it is an issue, I was intending to fix it as part of #3688 because `MultiWorld.can_beat_game()` uses almost identical sweeping code as `CollectionState.sweep_for_events()` so I'm making changes to `MultiWorld.can_beat_game()` too, and those changes will conflict with any current fix for this issue. This way I can put a "Fixes #3742" into the PR and it'll link to all the relevant information without having to add the entire description of this issue into the PR itself.

For such a minor issue, and given the backlog of PRs, I'm not sure it's worth creating a separate PR, but I can create one if preferred.
